django/db/migrations/writer.py

```diff
@@ -199,6 +198,10 @@ def as_string(self):
 
     @property
     def basedir(self):
+        # Defer import since `MigrationLoader` uses `MigrationRecorder` which
+        # declares `Migration` model.
+        from django.db.migrations.loader import MigrationLoader
+
```

Comment:
Have we explored making `MigrationRecorder.Migration` to be lazily created on the first class access instead?

Reply:
I think we should move https://github.com/django/django/blob/6713926ebe22172e50f283185f969275c326416d/django/db/migrations/recorder.py#L22 to a new file `django/django/db/migrations/migration.py` and import lazily here https://github.com/django/django/blob/6713926ebe22172e50f283185f969275c326416d/django/db/migrations/recorder.py#L44 and here https://github.com/django/django/blob/6713926ebe22172e50f283185f969275c326416d/django/db/migrations/recorder.py#L40 
What do you think?